Xamarin.Forms.Core/Shell/ShellUriHandler.cs

```diff
@@ -221,11 +230,6 @@ internal static List<RouteRequestBuilder> GenerateRoutePaths(Shell shell, Uri re
 			var routeKeys = Routing.GetRouteKeys();
 			for (int i = 0; i < routeKeys.Length; i++)
 			{
-				if (routeKeys[i] == originalRequest.OriginalString)
```

Comment:
This code just moved to the bottom of this method so we can try to match all the nested route parts